data_resource_api/factories/orm_factory.py

```diff
@@ -65,43 +65,74 @@ def evaluate_foreign_key(self, foreign_keys, field_name, field_type):
+            has_required_in_constraints = (
+                'constraints' in descriptor_field.keys() 
+                and 'required' in descriptor_field['constraints'].keys() 
+                and descriptor_field['constraints']['required']
+            )
+            return has_required_key_and_true or has_required_in_constraints
+
         sqlalchemy_fields = {}
+
         if isinstance(primary_key, str):
             primary_key = [primary_key]
-        for field in fields:
-            if ('required' in field.keys() and field['required']) or \
-                ('constraints' in field.keys() and 'required' in field['constraints'].keys() and field['constraints']['required']):
+
+        for descriptor_field in descriptor_fields:
+            if is_required(descriptor_field):
                 nullable = False
             else:
                 nullable = True
-            if field['name'] in primary_key:
-                sqlalchemy_fields[field['name']] = Column(
-                    self.get_sqlalchemy_type(field['type']), primary_key=True)
-            else:
-                is_foreign_key, reference_table = self.evaluate_foreign_key(
-                    foreign_keys, field['name'], field['type'])
-                if not is_foreign_key:
-                    sqlalchemy_fields[field['name']] = Column(
-                        self.get_sqlalchemy_type(field['type']), nullable=nullable)
-                else:
-                    try:
-                        sqlalchemy_fields[field['name']] = Column(
-                            self.get_sqlalchemy_type(
-                                field['type']), ForeignKey(reference_table, onupdate='CASCADE', ondelete='CASCADE'))
-                    except Exception as e:
-                        print('An exception occured {}'.format(e))
+            
+            # If this is the primary key
+            if descriptor_field['name'] in primary_key:
+                sqlalchemy_fields[descriptor_field['name']] = Column(
+                    self.get_sqlalchemy_type(descriptor_field['type']),
+                    primary_key=True
+                )
+                continue
+
+            # Otherwise check if its a foreign key
+            is_foreign_key, reference_table = self.evaluate_foreign_key(
+                foreign_keys,
+                descriptor_field['name'],
+                descriptor_field['type']
+            )
+            if is_foreign_key: 
+                try:
+                    sqlalchemy_fields[descriptor_field['name']] = Column(
+                        self.get_sqlalchemy_type(descriptor_field['type']),
+                        ForeignKey(
+                            reference_table,
+                            onupdate='CASCADE',
+                            ondelete='CASCADE'
+                        )
+                    )
+                except Exception as e:
```

Comment:
You reordered the logic flow, and now, we check for `is_foreign_key` before `not is_foreign_key`. I imagine that's fine....but your rearrangement highlights that we only catch exceptions for `is_foreign_key`. That seems odd, no? 

* It seems like we should be catching errors for all cases
* It would also be nice to not catch a naked exception
* It also seems like the exception handling should happen in `get_sqlalchemy_type`. 

I'd say open an issue for this.